From 40cc97cf77a4710e2cdbffd252fe7f741d030890 Mon Sep 17 00:00:00 2001
From: Robin Gong <b38343@freescale.com>
Date: Tue, 27 Jan 2015 10:27:36 +0800
Subject: [PATCH 0505/1594] MLK-10244-10 ARM: dts: imx7d: add snvs-power key
 support

commit e43e9a5ba6f965f0cc3bb1d0554b4c40ad0744bc from
git://git.freescale.com/imx/linux-2.6-imx.git

Enable rtc-snvs and snvs-power key, now we can use POWER key to wakeup
system.

Signed-off-by: Robin Gong <b38343@freescale.com>
(cherry picked from commit f292db590e0b01cf9580109c35d2a80a4b936c17)
---
 arch/arm/boot/dts/imx7d.dtsi |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/arch/arm/boot/dts/imx7d.dtsi b/arch/arm/boot/dts/imx7d.dtsi
index 3d1efd2..f166d05 100644
--- a/arch/arm/boot/dts/imx7d.dtsi
+++ b/arch/arm/boot/dts/imx7d.dtsi
@@ -346,6 +346,14 @@
 				};
 			};
 
+			snvs-pwrkey@0x30370000 {
+				compatible = "fsl, imx7d-snvs-pwrkey", "fsl,imx6sx-snvs-pwrkey";
+				reg = <0x30370000 0x10000>;
+				interrupts = <GIC_SPI 4 IRQ_TYPE_LEVEL_HIGH>;
+				fsl,keycode = <116>; /* KEY_POWER */
+				fsl,wakeup;
+			};
+
 			clks: ccm@30380000 {
 				compatible = "fsl,imx7d-ccm";
 				reg = <0x30380000 0x10000>;
-- 
1.7.5.4

